csharp/src/Ice/Reference.cs

```diff
@@ -609,27 +512,83 @@ public class FixedReference : Reference
+
+        internal override Reference Clone()
+        {
+            var clone = (base.Clone() as FixedReference)!;
+            clone._hashCode = 0;
+            return clone;
+        }
+
+        internal override bool IsRequestHandlerCached => true;
 
         internal FixedReference(Communicator communicator,
-                              Identity identity,
-                              string facet,
-                              InvocationMode mode,
-                              bool secure,
-                              Protocol protocol,
-                              Encoding encoding,
-                              Connection connection,
-                              int invocationTimeout,
-                              IReadOnlyDictionary<string, string> context,
-                              bool? compress)
-        : base(communicator, identity, facet, mode, secure, protocol, encoding, invocationTimeout, context)
+                                Identity identity,
+                                string facet,
+                                InvocationMode mode,
+                                bool secure,
+                                Protocol protocol,
+                                Encoding encoding,
+                                Connection connection,
+                                int invocationTimeout,
+                                IReadOnlyDictionary<string, string> context,
+                                bool? compress)
+            : base(communicator, identity, facet, mode, secure, protocol, encoding, invocationTimeout, context)
         {
             _fixedConnection = connection;
-            IsRequestHandlerCached = true;
+
             if (compress is bool compressValue)
             {
                 OverrideCompress = true;
                 Compress = compressValue;
             }
+            _requestHandler = CreateConnectionRequestHandler();
+        }
+
+        // Creates the ConnectionRequestHandler after various checks
+        private ConnectionRequestHandler CreateConnectionRequestHandler()
+        {
+            if (GetMode() == InvocationMode.Datagram)
+            {
+                if (!(_fixedConnection.Endpoint as Endpoint)!.Datagram())
+                {
+                    throw new ArgumentException("a fixed datagram proxy requires a datagram connection");
+                }
+            }
+            else if (GetMode() == InvocationMode.BatchOneway || GetMode() == InvocationMode.BatchDatagram)
+            {
+                throw new NotSupportedException("batch invocation modes are not supported for fixed proxies");
+            }
+
+            // If a secure connection is requested or secure overrides is set, check if the connection is secure.
+            bool secure;
+            DefaultsAndOverrides defaultsAndOverrides = Communicator.DefaultsAndOverrides;
+            if (defaultsAndOverrides.OverrideSecure)
```

Comment:
Shouldn't this use a nullable instead of OverrideSecure/OverrideSecureValue?

Reply:
Yes, I just kept the existing logic.